proxyd/consensus_poller.go

```diff
@@ -318,18 +336,22 @@ func (cp *ConsensusPoller) UpdateBackend(ctx context.Context, be *Backend) {
 	}
 
 	latestBlockNumber, latestBlockHash, err := cp.fetchBlock(ctx, be, "latest")
-	if err != nil {
-		log.Warn("error updating backend - latest block", "name", be.Name, "err", err)
+	if err != nil || latestBlockNumber == 0 {
+		log.Warn("error updating backend - latest block will not be updated", "name", be.Name, "err", err)
```

Comment:
we should also return from here, no point in trying to update the other fields after some have already failed

Reply:
any error on this method should prematurely return and not update the backend state

if the backend causes too much errors it will automatically breach the threshold of the intermittent errors and we dont need to explicit ban it

Reply:
Sure I can return from here. 

I implemented it this way so that the safe and finalized blocks could still advance if we wanted that

Reply:
I'm afraid of leaving edgy cases if we do partial updates, so lets avoid it entirely for now

it will be a full update or nothing